### ISSUE TITLE
quitting demo caused a panic

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#853]
Find and fix the cause of the panic that I got on my development machine when I quit with four players.

```
thread '<unnamed>' panicked at 'no candidates, amount: 200', mb2/src/table.rs:2344:17
```

```
Dealing #10103: 100 200 No-Limit Texas Hold’em (High only)
demo1472 antes 200 as the big blind
demo1655 blinds 100
demo1472 blinds 200
demo1515 quits in 4th place
demo1472 quits in 3rd place
demo1655 quits in 2nd place
Uncalled 100 returned to demo1472
```

This is probably due to returning uncalled action immediately (#716).

**Comment by ctm** [comment]
The Uncalled 100 is returned to demo1472, but demo1472 had already quit. So `return_uncalled` is not doing the right thing.

Before I implemented #716, `return_uncalled` was only called at the top of the showdown, but after `remove_excess_action_from_quit_players` was called. I'll have to double-check, but I think I can just put the call to `remove_excess_action_from_quit_players` inside `return_uncalled`.

**Comment by ctm** [comment]
If I revert commit `e62243c` (i.e. my fix for #716), the panic can still happen, so that was not the cause after all.

Since this bug has probably been there.a while and since a revert doesn't get rid of it, I'm not going to try to fix this in time for tonight's tournament. I'll poke around for a little while longer and then try to quiet my brain so I can get to sleep on time.

**Comment by ctm** [comment]
Turns out when I got rid of `orig_pots` as my fix for #844, I introduced this bug. It only happens when there are a bunch of quitters and a solo ante, so it was unlikely to show up in real life, nor did it show up in any of my testing because I didn't have enough players to trigger it.

Fixed. I'm going to merge it into master and deploy tomorrow since it's too close to tonight's game.
